src/state/ingress.py

```diff
@@ -39,11 +39,13 @@ class HAProxyBackend:
         backend_name: The name of the backend (computed).
         servers: The list of server each corresponding to a requirer unit.
         strip_prefix: Whether to strip the prefix from the ingress url.
+        redirect_https: Redirect incoming traffic to HTTPS.
     """
 
     backend_name: str
     servers: list[HAProxyServer]
     strip_prefix: bool = False
+    redirect_https: bool = False
```

Comment:
As enabling server-side encryption should have its own PR, we can probably remove this.